parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -222,18 +217,29 @@ setupCNIDirs() {
     chmod 755 $CNI_CONFIG_DIR
 }
 
-# For CNI/AzureCNI, we want to use the untar azurecni reference first. And if that doesn't exist on the vhd does the tgz?
-# And if tgz is already on the vhd then just untar into CNI_BIN_DIR
-# Latest VHD should have the untar, older should have the tgz. And who knows will have neither.
+
+# Reference CNI plugins is used by kubenet and the loopback plugin used by containerd 1.0 (dependency gone in 2.0)
+# The version used to be deteremined by RP/toggle but are now just hadcoded in vhd as they rarely change and require a node image upgrade anyways
+# Latest VHD should have the untar, older should have the tgz. And who knows will have neither. 
 installCNI() {
-    CNI_TGZ_TMP=${CNI_PLUGINS_URL##*/} # Use bash builtin ## to remove all chars ("*") up to the final "/"
+    #how do we keep this in sync with whats in AgentBaker/parts/linux/cloud-init/artifacts/components.json
+    # and how is that different from AgentBaker/vhdbuilder/packer/components.json 
+    #Only the seconde one caches arm? 
+    if [[ $(isARM64) == 1 ]]; then
+        CNI_TGZ_TMP="cni-plugins-linux-arm64-v1.4.1.tgz"
+    else 
+        CNI_TGZ_TMP="cni-plugins-linux-amd64-v1.4.1.tgz"
+    fi
+    
     CNI_DIR_TMP=${CNI_TGZ_TMP%.tgz}    # Use bash builtin % to remove the .tgz to look for a folder rather than tgz
 
     if [[ -d "$CNI_DOWNLOADS_DIR/${CNI_DIR_TMP}" ]]; then
-        mv ${CNI_DOWNLOADS_DIR}/${CNI_DIR_TMP}/* $CNI_BIN_DIR
+        #not clear to me when this would ever happen. assume its related to the line above Latest VHD should have the untar, older should have the tgz. 
+        mv ${CNI_DOWNLOADS_DIR}/${CNI_DIR_TMP}/* $CNI_BIN_DIR 
     else
         if [[ ! -f "$CNI_DOWNLOADS_DIR/${CNI_TGZ_TMP}" ]]; then
-            logs_to_events "AKS.CSE.installCNI.downloadCNI" downloadCNI
+            #how do I install a new error here? 
```

Comment:
you can add a new error code to cse_helpers.sh

Reply:
though, do we want to keep the install to make sure things still work even if there isn't an untar or targz?

we can probably just use the downloadURL from the components.json entry rather than relying on RP

Reply:
new error added. Seems garunteed that we will have unpacked based on install dependencies so not clear why we'd keep both code paths.